pkg/cloudprovider/aws/instance.go

```diff
@@ -219,12 +219,16 @@ func (p *InstanceProvider) getInstances(ctx context.Context, ids []*string) ([]*
 		return nil, fmt.Errorf("expected %d instance(s), but got %d", len(ids), len(describedInstances))
 	}
 	instances := []*ec2.Instance{}
-	for _, instance := range describedInstances {
-		if len(aws.StringValue(instance.PrivateDnsName)) == 0 {
-			err = multierr.Append(err, fmt.Errorf("got instance %s but PrivateDnsName was not set", aws.StringValue(instance.InstanceId)))
-			continue
+	if injection.GetOptions(ctx).AWSNodeNameConvention == "resource-name" {
+		instances = describedInstances
+	} else {
+		for _, instance := range describedInstances {
+			if len(aws.StringValue(instance.PrivateDnsName)) == 0 {
+				err = multierr.Append(err, fmt.Errorf("got instance %s but PrivateDnsName was not set", aws.StringValue(instance.InstanceId)))
```

Comment:
We should shortcircuit rather than checking all instances. I'm not sure the multierr is actually useful here:
```
if len(aws.StringValue(instance.PrivateDnsName)) == 0 {
  return nil, fmt.Errorf("got instance %s but PrivateDnsName was not set", aws.StringValue(instance.InstanceId)
}
```